python/ray/_raylet.pyx

```diff
@@ -718,7 +718,9 @@ cdef CRayStatus task_execution_handler(
                     traceback_str,
                     job_id=None)
                 sys.exit(1)
-        except SystemExit:
+        except SystemExit as e:
+            if not hasattr(e, "is_ray_terminate"):
+                logger.exception("SystemExit was raised from the worker")
```

Comment:
Longer term we should just handle the `__ray_terminate__` in the core worker

Reply:
Yeah we can do that. I think the user may be allowed to override it, but it seems like something that can be done with an explicit stop method as well.